tests/functional/test_send_reply.py

```diff
@@ -24,27 +24,35 @@ def test_send_reply_to_source(functional_test_logged_in_context, qtbot, mocker):
     def check_for_sources():
         assert len(list(gui.main_view.source_list.source_items.keys()))
 
+    # Select the last source in the source list to send a reply to
     qtbot.waitUntil(check_for_sources, timeout=TIME_RENDER_SOURCE_LIST)
     source_ids = list(gui.main_view.source_list.source_items.keys())
-    first_source_id = source_ids[0]
-    first_source_item = gui.main_view.source_list.source_items[first_source_id]
-    first_source_widget = gui.main_view.source_list.itemWidget(first_source_item)
-    qtbot.mouseClick(first_source_widget, Qt.LeftButton)
-    # Type something into the reply box and click the send button.
-    message = "Hello, world!"
+    last_source_item = gui.main_view.source_list.source_items[source_ids[-1]]
+    last_source_widget = gui.main_view.source_list.itemWidget(last_source_item)
+    qtbot.mouseClick(last_source_widget, Qt.LeftButton)
+    qtbot.wait(TIME_CLICK_ACTION)
     conversation = gui.main_view.view_layout.itemAt(0).widget()
-    # Focus on reply box text entry.
+
+    # Focus on the reply box and type a message
     qtbot.mouseClick(conversation.reply_box.text_edit, Qt.LeftButton)
-    # Type in a message to the reply box.
-    qtbot.keyClicks(conversation.reply_box.text_edit, message)
     qtbot.wait(TIME_CLICK_ACTION)
-    # Wait until the result of the click on the send button has caused the
-    # reply_sent signal to trigger.
+    qtbot.keyClicks(conversation.reply_box.text_edit, "Hello, world!")
+
+    # Wait until `reply_sent` signal is triggered
     with qtbot.waitSignal(conversation.reply_box.reply_sent):
         qtbot.mouseClick(conversation.reply_box.send_button, Qt.LeftButton)
-    qtbot.wait(TIME_CLICK_ACTION)
-    # Ensure the last widget in the conversation view contains the text we
-    # just typed.
+
+    # Ensure the last widget in the conversation view contains the text we just typed
     last_msg_id = list(conversation.conversation_view.current_messages.keys())[-1]
     last_msg = conversation.conversation_view.current_messages[last_msg_id]
-    assert last_msg.message.text() == message
+    assert last_msg.message.text() == "Hello, world!"
+
+    # Clean up: delete the last source with the reply
+    source_count = gui.main_view.source_list.count()
+    controller.delete_source(conversation.conversation_title_bar.source)
+
+    def check_source_list():
+        # Confirm there is now only one source in the client list.
```

Comment:
Nit: this comment's not strictly accurate.

Reply:
ah yeah these functional tests in general have a lot of inline comments that could be removed or updated - a good maintenance task for whenever we write or edit functional tests - too bad i missed this one

Reply:
ohhhh i see you want me to do that work as part of this pr - i can do that